crates/compiler/roar/src/storage.rs

```diff
@@ -0,0 +1,56 @@
+use roc_module::symbol::Symbol;
+use std::fmt::{Debug,Display};
+///The type used to identify registers
+pub(crate) type Id = u32;
+#[derive(Clone,Debug)]
+///A number of bytes an operation takes
+pub struct ByteSize(u32);
+///Equivalent to byte size, merely used as to specify the two
+pub(crate) type Offset = ByteSize;
+pub(crate) const WORD_SIZE : ByteSize = ByteSize(8);
+#[derive(Clone,Debug)]
+pub(crate) struct Register(pub Id);
+#[derive(Clone,Debug)]
+pub(crate) struct FloatRegister(pub Id);
+#[derive(Clone,Debug)]
+pub(crate) struct Global(Symbol);
+#[derive(Clone,Debug)]
+///The type of all literal values that fit inside a word
+pub enum LiteralValue {
+    Signed(i64),
+    Unsigned(u64),
+    Float(f64),
+    Char(char)
+}
+
+///The type of literal data, for instance strings 
+pub(crate) type LiteralData = Vec<u8>;
+#[derive(Clone,Debug)]
+///All valid outputs of an operation
+pub enum Output {
+    ///`_`, the null register, for things to be discarded (say, subtractions used as compares)
+    Null,
+    Register(Register),
+    FloatRegister(FloatRegister),
+    Global(Global)
```

Comment:
I don't think this is possible. Globals are always constants and can not be output to. You could have an output to a structured register/location on the stack though.